test/SgidClient.spec.ts

```diff
@@ -305,28 +376,105 @@ describe('SgidClient', () => {
         privateKey: 'malformed',
         redirectUri: MOCK_REDIRECT_URI,
         hostname: MOCK_HOSTNAME,
-        apiVersion: MOCK_API_VERSION,
       })
 
       await expect(
-        invalidPrivateKeyClient.userinfo(MOCK_ACCESS_TOKEN),
+        invalidPrivateKeyClient.userinfo({
+          accessToken: MOCK_ACCESS_TOKEN,
+        }),
       ).rejects.toThrow('Failed to import private key')
     })
 
     it('should throw when encrypted key is malformed', async () => {
       server.use(userInfoHandlerMalformedKey)
 
-      await expect(client.userinfo(MOCK_ACCESS_TOKEN)).rejects.toThrow(
-        'Unable to decrypt or import payload key',
-      )
+      await expect(
+        client.userinfo({
+          accessToken: MOCK_ACCESS_TOKEN,
+        }),
+      ).rejects.toThrow('Unable to decrypt or import payload key')
     })
 
     it('should throw when encrypted data is malformed', async () => {
       server.use(userInfoHandlerMalformedData)
 
-      await expect(client.userinfo(MOCK_ACCESS_TOKEN)).rejects.toThrow(
-        'Unable to decrypt payload',
+      await expect(
+        client.userinfo({
+          accessToken: MOCK_ACCESS_TOKEN,
+        }),
+      ).rejects.toThrow('Unable to decrypt payload')
+    })
+  })
+
+  describe('generatePkcePair', () => {
+    it('should generate a PKCE pair when no length is provided', () => {
+      const pkcePair = generatePkcePair()
+
+      expect(pkcePair.codeVerifier.length).toBe(43)
+      expect(pkcePair.codeChallenge.length).toBe(43)
+
+      expect(pkcePair.codeChallenge).toMatch(codeVerifierAndChallengePattern)
+      expect(pkcePair.codeVerifier).toMatch(codeVerifierAndChallengePattern)
+    })
+
+    it('should generate a PKCE pair of specified length when length between 43 (inclusive) and 128 (inclusive) is provided', () => {
+      for (let length = 43; length <= 128; length++) {
```

Comment:
omG

Reply:
how much time does this add to the tests? if it's cheap then no issue, but if this adding more than say 30s to the tests, I'm not sure it's worth it. we can just sample 3-4 sizes to test instead of testing every single number

Reply:
the entire test suite takes like ~1 second so i think it's fine :D